pkg/controller/deployment/deployment_controller.go

```diff
@@ -816,30 +834,102 @@ func (dc *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControl
 	if isPercent {
 		maxUnavailable = util.GetValueFromPercent(maxUnavailable, deployment.Spec.Replicas)
 	}
+
 	// Check if we can scale down.
```

Comment:
We need more explanation here about why what we're checking is the right thing to do. Please put the explanation from my PR comments into a code comment here.